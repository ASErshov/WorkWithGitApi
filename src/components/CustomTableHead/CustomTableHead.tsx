import React from 'react'
import { TableHead, TableRow, TableCell } from '@material-ui/core'

export type TableHeadProps = {
    lables: Array<{
        text:string,
        width?: number,
    }>
}

const CustomTableHead: React.FC<TableHeadProps> = ({lables}):React.ReactElement => {
    return(
        <TableHead>
          <TableRow>
            {lables.map((lable)=>(
            <TableCell align='center'  key={lable.text} width={lable.width?lable.width:60}>
                {lable.text}
            </TableCell>
            )
            )}
          </TableRow>
        </TableHead>
    )
}

export default CustomTableHead