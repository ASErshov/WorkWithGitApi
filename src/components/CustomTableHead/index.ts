export { default } from './CustomTableHead'
export * from './CustomTableHead'