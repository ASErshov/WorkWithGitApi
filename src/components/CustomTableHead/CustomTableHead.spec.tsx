import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import CustomTableHead from './index'

describe('<CustomTableHead />', () => {
  const Component = (
    <CustomTableHead lables={[{text:'first', width: 20}, {text:'second'}]}/>
  )

  it('Snapshot is correct', () => {
    const wrapper = renderer.create(Component).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('Passed props is valid', () => {
    const wrapper = mount(Component)

    expect(wrapper.props().lables).toEqual([{text:'first', width: 20}, {text:'second'}])
  
    wrapper.unmount()
  })
})
