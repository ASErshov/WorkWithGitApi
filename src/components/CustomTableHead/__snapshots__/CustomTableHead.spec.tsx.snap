// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTableHead /> Snapshot is correct 1`] = `
<thead
  className="MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
      scope="col"
      width={20}
    >
      first
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
      scope="col"
      width={60}
    >
      second
    </th>
  </tr>
</thead>
`;
